### ISSUE TITLE
`diff`/`patch` syntax highlighting doesn't work

### ISSUE THREAD

**Issue by twelho** [typst/typst#2727]
### Description

Typst doesn't syntax highlight diffs/patches, no matter what I try. No syntax highlighting is applied with `lang` set to `diff` or `patch`, and I even tried to import the [Diff.sublime-syntax](https://github.com/sublimehq/Packages/blob/ac11c3e99d25fdd443dd99a4ea5cb45bdf5b2957/Diff/Diff.sublime-syntax) directly using `syntaxes`, but that doesn't work either. I'm sure the diff itself is valid, as it is syntax-highlighted just fine in other editors, such as VS Code.

### Reproduction URL

https://typst.app/project/rdkysSRQ5EKL1OihIMWNmQ

**Comment by xalbd** [comment]
As far as I can tell, this is happening because the default theme that Typst provides for syntax highlighting does not provide colors for the scopes needed to highlight diffs. By adding some default colors, diffs can be highlighted. There doesn't seem to be any other underlying issue. 

With the typst code 
```typst 
#raw(lang: "patch", read("diff.patch"))
```
or
```typst 
#raw(lang: "diff", read("diff.patch"))
```

The changes present in https://github.com/xalbd/typst/commit/668e6aa7dd35e21bdae89d3368739da2ca698f8b produce the following result:

<img width="418" alt="Screenshot 2023-11-23 at 9 11 53 PM" src="https://github.com/typst/typst/assets/119540449/146897f5-64c0-4543-9668-ab15d07678b1">

**Comment by xalbd** [comment]
This issue has been fixed by #2838 and can be closed! 🎉
